libvirt.go

```diff
@@ -113,6 +113,30 @@ func (l *Libvirt) Capabilities() ([]byte, error) {
 	return []byte(caps), err
 }
 
+// called at connection time
+// it calls auth-list and authenticates if such auth type is supported
+func (l *Libvirt) authenticate() error {
+	resp, err := l.AuthList()
```

Comment:
nit: might be nice to keep the comment here about the AuthList being required even for AuthNone simply to protect any future code refactors from stumbling upon that requirement. 

> // libvirt requires that we call auth-list prior to connecting,
> // even when no authentication is used.